views/components/application/_theme_color_picker.erb

```diff
@@ -0,0 +1,18 @@
+<div id="<%= comp.id %>"
+     class="v-plugin v-plugin--color-picker v-input mdc-text-field mdc-text-field--outlined v-focusable"
+     style="height: auto; max-width: 100%;"
+     data-plugin-callback="ThemeColorPicker"
+     <% if comp.preview_data %>
+       data-preview-section-id= "<%= comp.preview_data[:section] %>"
+       data-preview-vars='<%= comp.preview_data[:set_vars].presence || [] %>'
```

Comment:
This is JSON, right? How/where is it being serialized to JSON, and does it need to be escaped?
```suggestion
       data-preview-vars='<%= h comp.preview_data[:set_vars].presence || [] %>'
```

Reply:
It is just an array, I think we should be fine. But it doesn't hurt